### PR TITLE
Support for global Hybrids and CAM functionals

### PR DESCRIPTION
**Pull request by vanderhe**
Accounts for recently introduced features in SkProgs (cf. corresponding [PR](https://github.com/dftbplus/skprogs/pull/30)). For now only basic functionality is provided (single-point, analytic forces, probably MD, ...).

Done/remaining tasks:
- [x] extend range-separated type by more general CAM functionals (alpha, beta, omega)
- [x] global hybrids (parse SK extra tag, only full-range Hartree-Fock evaluation)
- [x] check analytical forces against finite differences (global hybrids|CAM)
- [ ] documentation
- [ ] regression tests

Later PR's would have to address:
- periodic boundary conditions (+ eventually k-points)
- (real-time) TD-CAM-DFTB
- xTB
- ...

### CODE REVIEW

File: src/dftbp/dftbplus/inputdata.F90 [hunk 1]
```diff
@@ -108,6 +108,21 @@ module dftbp_dftbplus_inputdata
...
+    real(dp) :: camBeta
+
+    !> True, for global hybrids
+    logical :: tHyb
+
+    !> True, for long-range corrected functionals
+    logical :: tLc
+
+    !> True, for general CAM range-separation
+    logical :: tCam
```
Comment (awvwgk) [inline src/dftbp/dftbplus/inputdata.F90:124]: Can we make this an enumerator? Three independent logicals are hard to track, instead we could just use an enumerator (maybe `exchange_kernel`) with the values `none`, `hybrid`, `longRange`, and `cam` (or similar).
